lsusb.c

```diff
@@ -320,6 +322,59 @@ static void dump_device(
 	free(serial);
 }
 
+
+/* Dump only serial number as text (mode==0) or hexadecimal (mode!=0) */
+static void dump_serial(libusb_device *dev, struct libusb_device_descriptor *desc, int mode)
+{
+	libusb_device_handle *udev;
+	if(!desc->iSerialNumber) {
+		if(mode)	// In hexadecimal mode print a hint
+			printf("NONE");
+		else
+			if (verblevel > 0)
+				fprintf(stderr, "Device without serial number descriptor\n");
+		return;
+	}
+	if (libusb_open(dev, &udev)) {
+		fprintf(stderr, "Couldn't open device, some information will be missing\n");
+		return;
+	} 
+
+	if(mode) {
+		// For hexadecimal output
+		unsigned char buf[255];	// Ahhrrg! Took me several hours, that size > 256 not works correct :-(
+		unsigned char *ptr=buf;	// Seems to me, that there is a &255 somewhere, but length parameter is an int !
```

Comment:
I do not understand this comment, why?

Reply:
Hi Greg,
Comment is, because it took a lot time to find out that a buffer bigger then 255 can make problems with libusb_get_descriptor(). I read about the string descriptor and that first byte is the length. Normally I use maximum expected length + 1 for my buffers. So I used 256. But this function always fails then :-(. Length is declared as int, so I didn't expected.
Of course I understand, that you don't like it and want to remove. 
Was only a hint for other programmers to be careful with this parameter. Of course removing is OK.